test/framework/namespace.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2020 Red Hat, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package framework
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+)
+
+const (
+	namespaceLogFile = "logs/namespace_history.log"
+	fileFlags        = os.O_CREATE | os.O_WRONLY | os.O_APPEND
+	permissionMode   = 0666
+)
+
+// OnNamespacePostCreated triggers all post events for namespace is created.
+func OnNamespacePostCreated(namespace string) {
+	if err := addNamespaceToHistory(namespace); err != nil {
+		GetLogger(namespace).Warnf("Error updating namespace history %v", err)
+	}
+}
+
+// OnNamespacePostDeleted triggers all post events when namespace is deleted.
+func OnNamespacePostDeleted(namespace string) {
+	if err := removeNamespacesFromHistory(namespace); err != nil {
+		GetLogger(namespace).Warnf("Error removing namespace of history %v", err)
+	}
+}
+
+// GetNamespacesInHistory retrieves all the namespaces in the history.
+func GetNamespacesInHistory() []string {
+	input, err := ioutil.ReadFile(namespaceLogFile)
+	if err != nil {
+		// file does not exist
+		return []string{}
+	}
+
+	return strings.Split(string(input), "\n")
+}
+
+// ClearNamespaceHistory clears all the namespace history content.
+func ClearNamespaceHistory() {
+	os.Remove(namespaceLogFile)
+}
+
+func addNamespaceToHistory(namespace string) error {
+	file, err := os.OpenFile(namespaceLogFile, fileFlags, permissionMode)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+	if _, err = file.WriteString(fmt.Sprintf("%s\n", namespace)); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func removeNamespacesFromHistory(namespace string) error {
+	namespaces := GetNamespacesInHistory()
+	var newNamespaces []string
+	for _, oldNamespace := range namespaces {
+		if !strings.Contains(oldNamespace, namespace) {
```

Comment:
Wouldn't it be safer to use `==` ?

Reply:
As the namespace will contain the whole namespace, I thought it would make more sense to use the contains. But I agree with you, this is not necessary.
Changed to !=.